packages/invoices/src/OutboundShipment/api/hooks/line/useOutboundAllocateSelectedLines.ts

```diff
@@ -67,12 +67,45 @@ export const useOutboundAllocateSelectedLines = (): {
       return;
     }
 
-    const response = await mutateAsync(selectedUnallocatedLines);
-    const { allocateOutboundShipmentUnallocatedLines } = response;
-    const number = allocateOutboundShipmentUnallocatedLines?.length;
-    const onSuccess = success(t('messages.allocated-lines', { number }));
-    onSuccess();
-  };
+    const batchResponse = await mutateAsync(selectedUnallocatedLines);
+
+    if (batchResponse?.__typename === 'BatchOutboundShipmentResponse') {
+      const { allocateOutboundShipmentUnallocatedLines } = batchResponse;
+      const count = {
+        success: 0,
+        partial: 0,
+        failed: 0,
+      };
+
+      allocateOutboundShipmentUnallocatedLines?.forEach(line => {
+        const { id, response } = line;
```

Comment:
I'm not entirely enjoying the very specific, and nested, shape of returned objects. Have tried to be as safe as possible, though I wonder now if we may end up with no notifications when the api return shape is changed again. I _think_ that's better than notifying incorrectly?

Reply:
> I _think_ that's better than notifying incorrectly?

Yeah. Is it possible to have a general warning notification that always pops up whenever the return shape is different to what's expected?

Maybe only when in dev environment, if that's possible?  Just thinking that we'd want to notice this ASAP.

Reply:
perhaps bugsnag could catch it too?